### PR TITLE
Feature - Implement a read status for messages 

### PR DESCRIPTION
**Pull request by zhuyuanmao**
### Description
Implemented a new feature #2  where can track a read status for each message.  To achieve this, I created a new field `readStatus` on `Message` model and set `false` as default values. On the front end, I calculate the number of unread messages (`readStatus == false` and senders are not the logged-in user) , and then display it on the badge from the material ui.

## Screenshot
read-status-feature-screenshot

### CODE REVIEW

File: client/src/components/Sidebar/Chat.js [hunk 1]
```diff
@@ -39,11 +42,23 @@ class Chat extends Component {
...
       </Box>
     );
   }
 }
 
+const getUnreadMessagesCount = (conversation) => {
+  return conversation.messages.filter(
+    msg => msg.readStatus === false && msg.senderId === conversation.otherUser.id
+  ).length;
+}
```
Comment (0916dhkim) [inline client/src/components/Sidebar/Chat.js]: This calculates the unread count on every render, which is not ideal. Consider counting unread messages inside Redux reducer or on backend.

File: server/messenger_backend/models/message.py [hunk 2]
```diff
@@ -14,5 +14,6 @@ class Message(utils.CustomModel):
         related_name="messages",
         related_query_name="message"
     )
+    readStatus = models.BooleanField(default=False)
```
Comment (0916dhkim) [inline server/messenger_backend/models/message.py:17]: Good design choice :+1:

**Review by 0916dhkim — COMMENTED** [review]
@zhuyuanmao Good attempt so far. Here are things we can improve in this PR:
1. There should be a little avatar inside conversation indicating the last message read by the other user.
2. `latestMessageText` should be bold when a conversation is unread.
3. There is no logic for marking messages as read. When messages are read, the badge should disappear & the other user should know about the event in real time.

Please make necessary changes and let the reviewers know when this PR is ready for another review.

File: client/src/components/Sidebar/ChatContent.js [hunk 3]
```diff
@@ -18,21 +18,28 @@ const useStyles = makeStyles((theme) => ({
...
+  }
 }));
 
 const ChatContent = (props) => {
   const classes = useStyles();
 
   const { conversation } = props;
-  const { latestMessageText, otherUser } = conversation;
-
+  const { latestMessageText, otherUser, boldedPreviewText } = conversation;
```
Comment (FilipeGorodscy) [inline client/src/components/Sidebar/ChatContent.js]: I don't think you need a `boldedPreviewText` prop here. You could simply check if there are any unread messages for this particular conversation :)

File: client/src/store/utils/reducerFunctions.js [hunk 4]
```diff
@@ -16,7 +17,10 @@ export const addMessageToStore = (state, payload) => {
       const convoCopy = { ...convo };
       convoCopy.messages.push(message);
       convoCopy.latestMessageText = message.text;
-
+      if (convo.otherUser.id === message.senderId){
+        convoCopy.boldedPreviewText = true;
+        convoCopy.unreadMessagesCount = convo.unreadMessagesCount + 1;
```
Comment (FilipeGorodscy) [inline client/src/store/utils/reducerFunctions.js]:
```suggestion
        convoCopy.unreadMessagesCount += 1;
```

**Review by FilipeGorodscy — APPROVED** [review]
Great job addressing Danny's comments. I left some minor comments below. Feel free to merge this PR once you've addressed all changes :)

**Pull request merged**